teos/src/config.rs

```diff
@@ -225,13 +228,38 @@ impl Config {
                         _ => 8332,
                     }
                 }
-                Ok(())
+                network
             }
             Err(_) => {
-                Err(ConfigError(format!("btc_network not recognized. Expected {{bitcoin, testnet, signet, regtest}}, received {}",
-                self.btc_network)))
+                return Err(ConfigError(format!("btc_network not recognized. Expected {{bitcoin, testnet, signet, regtest}}, received {}",
+                self.btc_network)));
             }
+        };
+
+        // FIXME: This actually doesn't work. Network implementation is not there for rpc.
+        let real_btc_network = self
+            .get_bitcoin_rpc_client()?
+            .call_method::<Network>("getblockchaininfo", &[])
+            .await
+            .map_err(|e| ConfigError(format!("Call to getblockchaininfo failed: {}", e)))?;
+        if real_btc_network != teos_btc_network {
+            return Err(ConfigError(format!(
+                "btc_network conflict. bitcoind is running on {} but teosd is set to run on {}",
+                real_btc_network, teos_btc_network
+            )));
         }
+
+        Ok(())
+    }
+
+    /// Gets a Bitcoin RPC client.
+    pub fn get_bitcoin_rpc_client(&self) -> Result<RpcClient, ConfigError> {
+        let http_endpoint =
+            HttpEndpoint::for_host(self.btc_rpc_connect.to_owned()).with_port(self.btc_rpc_port);
+        let rpc_credentials =
+            base64::encode(&format!("{}:{}", self.btc_rpc_user, self.btc_rpc_password));
+        RpcClient::new(&rpc_credentials, http_endpoint)
+            .map_err(|e| ConfigError(format!("Failed to connect to bitcoind: {}", e)))
```

Comment:
This is copied from [here](https://github.com/talaia-labs/rust-teos/blob/79fc2d37e87795a38bd87b166f715461403ff8a9/src/bitcoin_cli.rs#L97-L102).

Reply:
It's very weird that https://github.com/talaia-labs/rust-teos/blob/79fc2d37e87795a38bd87b166f715461403ff8a9/src/bitcoin_cli.rs#L97-L102 was never used in the codebase and yet clippy didn't error on it for being deadcode :confused:.

Reply:
It's all public, isn't it? Public code does not get marked as deadcode as it can be used by other libs.

Reply:
Oh, right that makes sense.